sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -207,6 +252,36 @@ private PartitionWriter(TopicPartition tp, long firstOffset) throws IOException
               firstOffset,
               GZIPChunkThreshold,
               format.init(tp.topic(), tp.partition(), firstOffset));
+      lastFlush = Instant.now().toEpochMilli();
+    }
+
+    public BlockGZIPFileWriter getWriter() {
+      return writer;
+    }
+
+    public boolean shouldBeDone() {
+      long timeSinceLastFlush = Instant.now().toEpochMilli() - lastFlush;
+      boolean doPeriodicFlush = timeSinceLastFlush >= flushIntervalMs;
+      if (doPeriodicFlush) {
+        log.debug("{} performing a periodic flush on {}", name(), tp);
+      }
```

Comment:
Why don't we early-return from here? If we know that a periodic flush should be done, what's the point in calculating and logging the size-based flush details?